lib/convict.js

```diff
@@ -731,8 +731,8 @@ convict.addParser = function(parsers) {
 
   parsers.forEach(function(parser) {
     if (!parser) throw new Error('Invalid parser');
-    if (!parser.extension) throw new Error('Missing parser extension');
-    if (!parser.parse) throw new Error('Missing parser parse function');
+    if (!parser.extension) throw new Error('Missing parser.extension');
+    if (!parser.parse) throw new Error('Missing parser.parse function');
 
     if (typeof parser.parse !== 'function') throw new Error('Invalid parser.parse function');
```

Comment:
I think we should not care with type checks in this context. Adding a parser will fail if the type is not right. Adding type checks is worthy in JavaScript when a situation becomes critical or hard to debug. This is not the case here.

Reply:
The common consensus is that it should fail if the **parse** function is not a function, either immediately here, or on the next call to **loadFile**.

I still prefer that it fails immediately, as it helps detecting errors sooner. Imagine this code, without immediate type checking there's always a chance for errors to slip into production.
```
// Which parsers.getYamlParser() return undefined
convict.addParser({ extension: 'yml', parse: parsers.getYamlParser() });

const conf = convict({ ... });
if (process.env.NODE_ENV == 'production') convict.loadFile('/etc/config/app.yml');
```
Beside, **addParser** should only be call a few times on startup. So type checking shouldn't hurt any performance.

Reply:
OK